platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/event/preconsensus/SyncPreConsensusEventWriterTests.java

```diff
@@ -134,13 +127,22 @@ void standardOperationTest() throws IOException, InterruptedException {
         writer.start();
 
         long minimumGenerationNonAncient = 0;
-        for (final EventImpl event : events) {
+        final Iterator<EventImpl> iterator = events.iterator();
+        while (iterator.hasNext()) {
+            final EventImpl event = iterator.next();
+
             sequencer.assignStreamSequenceNumber(event);
 
             minimumGenerationNonAncient =
                     Math.max(minimumGenerationNonAncient, event.getGeneration() - generationsUntilAncient);
             writer.setMinimumGenerationNonAncient(minimumGenerationNonAncient);
 
+            if (event.getGeneration() < minimumGenerationNonAncient) {
+                // Although it's not common, it's actually possible that the generator will generate
```

Comment:
Would it make sense to extend the functionality of the generator, to be able to specify min generation?

Reply:
The main problem with that strategy is that the minimum generation non-ancient increases over time and is tracked externally to the generator, meaning that the generator has no knowledge of it.